### ISSUE TITLE
Set zoom level on markers update

### ISSUE THREAD

**Issue by mycarrysun** [SCODEMeetup/community-services-locator-ui#51]
Going to need some help on this with someone who understands the HOC convention, it is a little over my head coming from Vuejs.

Would like to use [this example](https://www.reonomy.com/blog/post/offsetting-bounds-and-zoom-on-a-google-map) to center and zoom the map when a set of locations is loaded as markers on the map. I have use in another project and it works great and is light on CPU.

What I need help with is making the `markers` prop on the `MapClass` component observable, so we can attach a handler anytime it changes and loop through to create a new bounds object which would then be able to set the zoom level.

Would appreciate anyone's expertise here who knows how to use the `recompose` library!

Side note: the [`recompose`](https://github.com/acdlite/recompose) library is at [EOL](https://github.com/acdlite/recompose/issues/756#issuecomment-438674573) and will not be receiving future updates which can be a security concern if there are vulnerabilities...maybe we should consider removing this library to accommodate and go more vanilla React?

**Comment by mihiramin89** [comment]
Yes I noticed the recompose library was EOL. Would love to change this to something that better follows react components. I do not particularly like putting majority of the code in one file. 

I can try to take a stab at it in the next week or so. 

Mihir

**Comment by mycarrysun** [comment]
@mihiramin89 excellent thanks for the help sir!

**Comment by downtownbob** [comment]
Taking this card - will reach out with any questions.
